solr/solrj-zookeeper/src/test/org/apache/solr/common/cloud/SolrZkClientTest.java

```diff
@@ -281,4 +282,42 @@ public void testCheckInterrupted() {
     SolrZkClient.checkInterrupted(new InterruptedException());
     assertTrue(Thread.currentThread().isInterrupted());
   }
+
+  @Test
+  public void testWithSolrResourceLoader() {
+    setupSystemProperties();
+    SolrResourceLoader solrResourceLoader =
+        new SolrResourceLoader(Path.of("."), ClassLoader.getSystemClassLoader());
+    new SolrZkClient.Builder()
+        .withUrl(zkServer.getZkHost())
+        .withTimeout(AbstractZkTestCase.TIMEOUT, TimeUnit.MILLISECONDS)
+        .withSolrClassLoader(solrResourceLoader)
+        .build()
+        .close(); // no more tests needed. We only test class instantiation
+    clearSystemProperties();
+  }
+
+  private static void setupSystemProperties() {
```

Comment:
[0] Small nit: this might benefit from a name that describes what sysprops you're setting up. e.g. `enableCustomCredentialsProvider` or something similar.

But if you like that name as-is, that's also fine.

Reply:
done :)